server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/SearchTemplatesAction.java

```diff
@@ -107,11 +108,18 @@ public void handle(Request wsRequest, Response wsResponse) throws Exception {
 
   private static void buildDefaultTemplatesResponse(SearchTemplatesWsResponse.Builder response, SearchTemplatesData data) {
     TemplateIdQualifier.Builder templateUuidQualifierBuilder = TemplateIdQualifier.newBuilder();
-    for (DefaultPermissionTemplateFinder.TemplateUuidQualifier templateUuidQualifier : data.defaultTemplates()) {
-      response.addDefaultTemplates(templateUuidQualifierBuilder
-        .clear()
-        .setQualifier(templateUuidQualifier.getQualifier())
-        .setTemplateId(templateUuidQualifier.getTemplateUuid()));
+
+    DefaultTemplatesResolverImpl.ResolvedDefaultTemplates resolvedDefaultTemplates = data.defaultTemplates();
+    response.addDefaultTemplates(templateUuidQualifierBuilder
+      .setQualifier(Qualifiers.PROJECT)
+      .setTemplateId(resolvedDefaultTemplates.getProject()));
+
+    if (resolvedDefaultTemplates.getView().isPresent()) {
+      response.addDefaultTemplates(
+        templateUuidQualifierBuilder
+          .clear()
+          .setQualifier(Qualifiers.VIEW)
+          .setTemplateId(resolvedDefaultTemplates.getView().get()));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)